src/wollok/lang.wlk

```diff
@@ -932,14 +930,20 @@ class Collection {
   /**
    * Answers the concatenated string representation of the elements in the given set.
    * You can pass an optional character as an element separator (default is ",")
+   *
+   * Example:
+   *      ["hola", "como", "estas"].join(" ") ==> Answers "hola como estas"
    */
   method join(separator) =
     if (self.isEmpty()) ""
-    else self.subList(1).fold(self.first().toString(), { string, element => string + separator + element.toString() })
+    else self.asList().subList(1).fold(self.asList().first().toString(), { string, element => string + separator + element.toString() })
```

Comment:
I know it's not a one-liner, but maybe it would be faster if we do the `self.asList()` just once.

Reply:
Yeap. Let's do it.